pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatPatternSpec.java

```diff
@@ -52,7 +66,28 @@ public DateTimeFormatPatternSpec(String timeFormat, String sdfPatternWithTz) {
         String timezoneString = m.group(TIMEZONE_GROUP).trim();
         _dateTimeZone = DateTimeZone.forTimeZone(TimeZone.getTimeZone(timezoneString));
       }
+      validateSdfPattern(_sdfPattern, requireLexicographicOrdering);
       _dateTimeFormatter = DateTimeFormat.forPattern(_sdfPattern).withZone(_dateTimeZone).withLocale(DEFAULT_LOCALE);
+
+    }
+  }
+
+  private static void validateSdfPattern(String sdfPattern, boolean requireLexicographicOrdering) {
```

Comment:
Let's move this into the `DateTimeUtils` as a general util method. I don't think we need a flag for `requireLexicographicOrdering`, it should always be checked

Reply:
Can we also check if the sdf is valid?

Reply:
actually I can't make it into DateTimeUtils because this class is in pinot-spi and DateTimeUtils is in pinot-common

Reply:
We can move it into `DateTimeUtils` and move the validation into the `SchemaUtils.validate()`. The validation logic should not be within the spi

Reply:
hmm i dont think this will work: SchemaUtils.validate only validates Schema. this means either
1. we disallow all non-lexicographic ordering on date/time column at all
2. we still cant tell whether a non-lexicographic ordering date/time column will cause problem in segment generation because the info is in TableConfig.

The only place I can think of that make sense is TableConfigUtils.validate(tableconfig, schema).
but it is not guarantee to be validated because the API said schema is nullable and if null, schema related checks are skipped.

Reply:
We should disallow all non-lexicographic date time format as it won't work properly. The range query will not follow the time ordering, which is in most cases unexpected if not all. IMO it is safe to add this validation to the schema.

Reply:
hmm. this is concerning. because we are literally telling the user any `mm/dd/YYYY` formatted raw data is not importable to pinot. (if want to declared as date-time format).

Reply:
In order to ingest such column as time column, it has to be converted to lexicographic order to prevent unexpected query result. It can be done via the ingestion transform. We should make it clear in our documentation

Reply:
done. but I still put the logic in spi, alongside with the other DateTimeFieldSpec validate methods. (we need those to run first before we can determine the schema is actually simple date format). 

for now my goal is to make this bug fix. later I can follow up with the validate code refactor out of SPI into common.